File: exercises/exercise-1-testing-java/src/main/java/io/miragon/example/adapter/outbound/zeebe/newsletter/NewsletterSubscriptionProcessAdapter.java
package io.miragon.example.adapter.outbound.zeebe.newsletter;

import io.miragon.example.adapter.process.config.ProcessEngineApi;
import io.miragon.example.adapter.process.generated.NewsletterSubscriptionProcessApi;
import io.miragon.example.application.port.outbound.newsletter.NewsletterSubscriptionProcess;
import io.miragon.example.domain.SubscriptionId;
import org.springframework.stereotype.Component;

import java.util.Map;

@Component
public class NewsletterSubscriptionProcessAdapter implements NewsletterSubscriptionProcess {

    private final ProcessEngineApi engineApi;

    public NewsletterSubscriptionProcessAdapter(ProcessEngineApi engineApi) {
        this.engineApi = engineApi;
    }

    @Override
    public long submitForm(SubscriptionId id) {
        Map<String, Object> variables = Map.of("subscriptionId", id.value().toString());
        return engineApi.startProcess(
            NewsletterSubscriptionProcessApi.PROCESS_ID,
            variables
        );
    }

    @Override
    public void confirmSubscription(SubscriptionId id) {
        engineApi.sendMessage(
            NewsletterSubscriptionProcessApi.Messages.MESSAGE_SUBSCRIPTION_CONFIRMED,
            id.value().toString()
        );
    }

}

File: exercises/exercise-1-testing-kotlin/src/main/kotlin/io/miragon/example/adapter/outbound/zeebe/newsletter/NewsletterSubscriptionProcessAdapter.kt
package io.miragon.example.adapter.outbound.zeebe.newsletter

import io.miragon.example.adapter.process.config.ProcessEngineApi
import io.miragon.example.adapter.process.generated.NewsletterSubscriptionProcessApi
import io.miragon.example.application.port.outbound.newsletter.NewsletterSubscriptionProcess
import io.miragon.example.domain.SubscriptionId
import org.springframework.stereotype.Component

@Component
class NewsletterSubscriptionProcessAdapter(
    private val engineApi: ProcessEngineApi
) : NewsletterSubscriptionProcess {

    override fun submitForm(id: SubscriptionId): Long {
        val variables = mapOf("subscriptionId" to id.value.toString())
        return engineApi.startProcess(
            processId = NewsletterSubscriptionProcessApi.PROCESS_ID,
            variables = variables
        )
    }

    override fun confirmSubscription(id: SubscriptionId) {
        engineApi.sendMessage(
            messageName = NewsletterSubscriptionProcessApi.Messages.MESSAGE_SUBSCRIPTION_CONFIRMED,
            correlationId = id.value.toString(),
        )
    }
}